inc/render/class-posts-grid-block.php

```diff
@@ -350,26 +307,111 @@ protected function render_featured_post( $post, $attributes ) {
-			unset( $args['post_type'] );
+		$offset = ! empty( $attributes['offset'] ) ? $attributes['offset'] : 0;
 
-			$products = wc_get_products( $args );
+		$categories = 0;
 
-			// convert to array of post objects since the rest of the code expects that.
-			return array_map(
-				function( $product ) {
-					return $product->get_id();
-				},
-				$products
-			);
+		if ( isset( $attributes['categories'] ) ) {
+			$cats = array();
+
+			foreach ( $attributes['categories'] as $category ) {
+				$cats[] = $category['id'];
+			}
+
+			$categories = join( ', ', $cats );
+		}
+
+		$args = array(
+			'post_type'        => $attributes['postTypes'],
+			'posts_per_page'   => $attributes['postsToShow'],
+			'post_status'      => 'publish',
+			'order'            => $attributes['order'],
+			'orderby'          => $attributes['orderBy'],
+			'offset'           => $offset,
+			'category'         => $categories,
+			'suppress_filters' => false,
+			'no_found_rows'    => true,
+		);
+
+		if ( $count_posts ) {
+			$args['offset']        = $args['posts_per_page'] * ( $page_number - 1 ) + $args['offset'];
+			$args['no_found_rows'] = false;
+			$args['paged']         = $page_number;
+		}
+
+		// Handle the case when the post type is a WooCommerce product.
+		if ( isset( $args['post_type'] ) && in_array( 'product', $args['post_type'] ) && function_exists( 'wc_get_products' ) ) {
+
+			if ( isset( $attributes['categories'] ) ) {
+
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+				$args['tax_query'] = array();
+				foreach ( $attributes['categories'] as $category ) {
+					if ( isset( $category['slug'] ) ) {
+						$args['tax_query'][] = array(
+							'taxonomy' => 'product_cat',
+							'field'    => 'slug',
+							'terms'    => $category['slug'],
+						);
+					}
+				}
+				$args['tax_query']['relation'] = 'OR';
+			}
+		}
+
+		$args = apply_filters(
+			'themeisle_gutenberg_posts_block_query',
+			$args,
+			$attributes
+		);
+
+		$query = new \WP_Query( $args );
+
+		if ( $count_posts ) {
+			$total_posts += $query->max_num_pages;
+		}
+
+		return $query->posts;
+	}
+
+	/**
+	 * Render the pagination.
+	 *
+	 * @param int $page_number The page number.
+	 * @param int $total_pages The total pages.
+	 * @return string
+	 */
+	protected function render_pagination( $page_number, $total_pages ) {
+		// Check permalink structure.
+		$permalink_structure = get_option( 'permalink_structure' );
+		if ( $permalink_structure ) {
+			$format = 'page/%#%/';
+		} else {
+			$format = '&paged=%#%';
 		}
+		$base = get_pagenum_link( 1 ) . '%_%';
+
+		$output  = '<div class="o-posts-grid-pag">';
+		$output .= paginate_links(
+			array(
+				'base'      => $base,
+				'format'    => $format,
+				'current'   => $page_number,
+				'total'     => $total_pages,
+				'prev_text' => __( 'Prev', 'otter-blocks' ),
+				'next_text' => __( 'Next', 'otter-blocks' ),
+			)
+		);
+		$output .= '</div>';
```

Comment:
@HardeepAsrani, do you see something wrong with this? When added to Home page, the link is something like this `https://cycleadvertisement.s2-tastewp.com/page/2/` and is not working. If you put the block on a post page, the link will be like `https://cycleadvertisement.s2-tastewp.com/index.php/2023/10/12/post/page/2/` and is working.

Reply:
@Soare-Robert-Daniel When you're going to that URL, I suppose it's redirecting you to 404? Or is it still opening the home?

Also, is it the sam eissue with Qubely?

Reply:
Might be some issue with nginx/htaccess configs maybe, then we can skip that part and live with it for this version.